### PR TITLE
chore: update manyToMany update form golden file

### PR DESCRIPTION
**Pull request by zchenwei**
*Description of changes:*

- `manyToMany` allows linking duplicate records. Update golden file to give the same experience as UX in console.
- move to its own folder and add a new schema separate from create form

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-create-form-with-many-to-many-relationship/TagUpdateForm.jsx [hunk 1]
```diff
@@ -277,38 +277,69 @@ export default function TagUpdateForm(props) {
           modelFields = onSubmit(modelFields);
         }
         try {
-          const postsToLink = [];
-          const postsToUnlink = [];
-          const postsSet = new Set();
-          const linkedPostsSet = new Set();
-          Posts.forEach((r) => postsSet.add(r.id));
-          linkedPosts.forEach((r) => linkedPostsSet.add(r.id));
+          const postsToLinkMap = new Map();
```
Comment (zchenwei) [inline packages/codegen-ui-golden-files/lib/react/forms/datastore-create-form-with-many-to-many-relationship/TagUpdateForm.jsx:280]: Using map instead to keep track of the count

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx [hunk 2]
```diff
@@ -277,38 +277,69 @@ export default function TagUpdateForm(props) {
...
-          Posts.forEach((r) => postsSet.add(r.id));
-          linkedPosts.forEach((r) => linkedPostsSet.add(r.id));
+          const postsToLinkMap = new Map();
+          const postsToUnLinkMap = new Map();
+          const postsMap = new Map();
+          const linkedPostsMap = new Map();
+
+          Posts.forEach((r) => {
+            const count = postsMap.get(r.id);
+            const newCount = count ? count + 1 : 1;
```
Comment (cwoolum) [inline packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:287]: Why are we adding one to the count?

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx [hunk 3]
```diff
@@ -277,38 +277,69 @@ export default function TagUpdateForm(props) {
...
+                postsToLinkMap.set(id, diffCount);
+              }
+            } else {
+              postsToLinkMap.set(id, count);
             }
           });
 
           const promises = [];
-          promises.push(...postsToUnlink.map((post) => DataStore.delete(TagPost, (r) => r.postID.eq(post.id))));
+          postsToUnLinkMap.forEach(async (count, id) => {
```
Comment (cwoolum) [inline packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:322]: Can we just do a `flatMap` with this?

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx [hunk 4]
```diff
@@ -277,38 +277,69 @@ export default function TagUpdateForm(props) {
...
-            ...postsToLink.map((post) =>
-              DataStore.save(
-                new TagPost({
-                  tag: tagRecord,
-                  post,
-                }),
-              ),
-            ),
-          );
+          postsToLinkMap.forEach((count, id) => {
```
Comment (cwoolum) [inline packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:331]: Could do a `flatMap` here also and spread it into the `promises` array

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx [hunk 5]
```diff
@@ -277,38 +277,69 @@ export default function TagUpdateForm(props) {
...
           });
 
           const promises = [];
-          promises.push(...postsToUnlink.map((post) => DataStore.delete(TagPost, (r) => r.postID.eq(post.id))));
+          postsToUnLinkMap.forEach(async (count, id) => {
+            const tagPostRecords = await DataStore.query(TagPost, (r) =>
+              r.and((r) => [r.post.id.eq(id), r.tag.id.eq(tagRecord.id)]),
+            );
+            for (let i = 0; i < count; i++) {
+              promises.push(DataStore.delete(tagPostRecords[i]));
```
Comment (bombguy) [inline packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:327]: Have you considered the limit to # of promises for the Promise.all? I'm thinking when we are trying to update a tag that has 1000+ post that are linked, we may run into problems. Do we have ability to batch these requests?

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx (hunk 2 above)
Comment (zchenwei) [reply packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:287]: According to the current UX, it allows adding duplicate records to the join table so have to track the count

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx (hunk 3 above)
Comment (zchenwei) [reply packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:322]: `postsToUnLinkMap` is a hasmap, I do not think we can do that

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx (hunk 4 above)
Comment (zchenwei) [reply packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:331]: `postsToLinkMap` is a hashmap, I do not think we can do that

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx (hunk 3 above)
Comment (cwoolum) [reply packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:322]: Aaah, that's right. Thanks!

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx (hunk 5 above)
Comment (zchenwei) [reply packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx:327]: Let's take it as an improvement and add it later considering the timeline

**Review by cwoolum — APPROVED** [review]

**Review by letsbelopez — APPROVED** [review]

**Pull request merged**